src/string.rs

```diff
@@ -125,112 +123,89 @@ pub fn string_regex_parsed(expr: &rs::Expr)
-               .sboxed())
-        },
-        LiteralBytes { ref bytes, casei: false } =>
-            Ok(Just(bytes.to_owned()).sboxed()),
-        LiteralBytes { ref bytes, casei: true } => {
-            let bytes = bytes.to_owned();
-            Ok(bits::bitset::between(0, bytes.len())
-               .prop_map(move |cases|
-                         cases.into_bit_vec().iter().zip(bytes.iter())
-                         .map(|(case, &byte)| flip_ascii_case(case, byte))
-                         .collect::<Vec<_>>()).sboxed())
+    use self::rs::hir::{Class, HirKind};
+
+    match *hir.kind() {
+        HirKind::Empty => Ok(Just(vec![]).sboxed()),
+        HirKind::Literal(ref l) => {
+            match *l {
+                rs::hir::Literal::Unicode(ref ch) => {
+                    Ok(Just(ch.to_string()
+                        .into_bytes()).sboxed())
+                },
+                rs::hir::Literal::Byte(ref byte) => {
+                    Ok(Just(::core::iter::once(*byte).collect()).sboxed())
+                }
+            }
         },
 
-        AnyChar => Ok(char::any().prop_map(to_bytes).sboxed()),
-        AnyCharNoNL => {
-            static NONL_RANGES: &[(char,char)] = &[
-                ('\x00', '\x09'),
-                // Multiple instances of the latter range to partially make up
-                // for the bias of having such a tiny range in the control
-                // characters.
-                ('\x0B', ::std::char::MAX),
-                ('\x0B', ::std::char::MAX),
-                ('\x0B', ::std::char::MAX),
-                ('\x0B', ::std::char::MAX),
-                ('\x0B', ::std::char::MAX),
-            ];
-            Ok(char::ranges(Cow::Borrowed(NONL_RANGES))
-               .prop_map(to_bytes).sboxed())
-        },
-        AnyByte => Ok(num::u8::ANY.prop_map(|b| vec![b]).sboxed()),
-        AnyByteNoNL => Ok((0xBu8..).sboxed()
-                          .prop_union((..0xAu8).sboxed())
-                          .prop_map(|b| vec![b]).sboxed()),
-
-        Class(ref class) => {
-            let ranges = (**class).iter().map(
-                |&rs::ClassRange { start, end }| (start, end)).collect();
-            Ok(char::ranges(Cow::Owned(ranges))
-               .prop_map(to_bytes).sboxed())
+        HirKind::Class(ref c) => {
+            match c {
+                Class::Unicode(ref class) => {
+                    let ranges = (class).iter().map(
+                        |r: &rs::hir::ClassUnicodeRange | (r.start(), r.end())).collect();
+                    Ok(char::ranges(Cow::Owned(ranges))
+                        .prop_map(to_bytes).sboxed())
+                },
+                Class::Bytes(ref class) => {
+                    let subs = (class).iter().map(
+                        |r: &rs::hir::ClassBytesRange | if 255u8 == r.end() {
+                            (r.start()..).sboxed()
+                        } else {
+                            (r.start()..r.end()).sboxed()
```

Comment:
Should this be `r.end() + 1`? (Though I notice the old code also just uses `start..end`)